packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -733,6 +742,66 @@ export async function launch({
         return true;
       }
 
+      case ActionType.ENACTED_UPGRADE: {
+        // Install and execute new core proposals.
+        const {
+          upgradePlan: { info: upgradeInfoJson = null } = {},
+          blockHeight,
+          blockTime,
+        } = action;
+        const upgradePlanInfo = upgradeInfoJson && JSON.parse(upgradeInfoJson);
```

Comment:
It would be useful to go thru something like `parseLocatedJson` so that errors indicate where the syntax error came from.